vector/any.go

```diff
@@ -21,3 +23,23 @@ type Puller interface {
 }
 
 type Builder func(*zcode.Builder) bool
+
+func Combine(desiredLen uint32, vec Any, index []uint32, add Any) Any {
```

Comment:
Nit: I don't think desiredLen is necessary since I think it's always equal to vec.Len()+uint32(len(index)).

Reply:
It is necessary the way it's being used in index expressions but perhaps I am approaching this wrong. Happy to talk about this.